pennylane/ops/qubit/observables.py

```diff
@@ -435,13 +435,20 @@ class BasisStateProjector(Projector, Operation):
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
-        wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
+        wires = qml.wires.Wires(wires)
 
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        if state is None:
```

Comment:
I don't think we particularly need a check for whether or not the state is `None`.

Our philosophy on errors is:

* Is this a mistake that is hard to debug if someone makes it?
* Is it a common mistake that users have been complaining about?
* Is it something that a user might expect to be able to do, but we just haven't implemented yet?

For something like `Projector(None, wires=0)`, I think we can just let that fall over some time later in the pipeline.

Basically, we can't check everything, and checks slow things and add complexity, so only add them when they really contribute something necessary.

Reply:
done, please review it and let me know.